javascript/javascript1/week2/flight-booking.js

```diff
@@ -0,0 +1,45 @@
+//Flight booking fullname function
+
+function getFullName (firstName, surName) {
+    return firstName + " " + surName;
+  }
+  
+  console.log (getFullName ("Benjamin", "Hughes")); // returns "Benjamin Hughes"
+  
+  //Formal fullname
+  
+  function getFullName (firstName, surName, useFormalName) {
+    if (useFormalName === true) {
+      console.log ("Lord" + " " + firstName + " " + surName); 
+    } else if (useFormalName === false) {
+        console.log (firstName + " " + surName); 
+      } else 
+        console.log ("login-failed")
+    }
+
+    //What if the person is a woman?
+
+function getFullName (firstName, surName, useFormalName, gender) {
+    if (gender === "woman" && useFormalName === true) {
+      console.log ("Lady" + " " + firstName + " " + surName);
+    } else if (gender === "woman" && useFormalName === false) {
+      console.log (firstName + surName);
+    } else if ( gender === "man" && useFormalName === true) {
+      console.log ("Lord" + firstName + " " + surName);
+    } else if (gender === "man" && useFormalName === false) {
+      console.log (firstName + surName);
+    } else
+      console.log ("login-failed");
+  }
```

Comment:
When you define multiple functions with the same the last one will replace or override all functions before.
Therefore create only one function does all things that requested